### ISSUE TITLE
Assigning elements to vector beyond its length does not have an effect

### ISSUE THREAD

**Issue by sebffischer** [dgkf/R#75]
```r
x <- true
x[2] <- NA
#> [1] true
```

**Comment by sebffischer** [comment]
The question is what the desired behavior here is. 
I think that this should just throw an error.

The only scenario where this seems useful is to do something like below

```r
x <- 1:10
x[11] = 11
x[12:13] = 12:13
```

However, this can easily be achieved with 

```r
x <- 1:10
x <- c(x, 1)
x <- c(x, 11:12)
```

Something like below makes rather little sense I guess. 

```r
x <- 1:10
x[13] <- 1
```

**Comment by dgkf** [comment]
I don't think `c()` is ideal because it (at least in R) does some weird coercion stuff. I see that as primarily an issue with `c()`, though, and otherwise think a means of appending to a vector in a more type-stable would be a preferred solution.

I _could_ see a scenario where you want to create a sparse vector, but can't determine the length initially:

```r
vec <- character(0)
vec[10000] <- "needle"
```

That said, I think the cases for this behavior are quite rare and the preferred approach would be to initialize or grow a vector of appropriate length and only then populate it.

So in short, I'd be fine with this becoming an error.
